qiita_db/sql_connection.py

```diff
@@ -504,140 +410,212 @@ def execute_fetchone(self, sql, sql_args=None):
+        From psycopg2 documentation, only variable values should be bound
         via sql_args, it shouldn't be used to set table or field names. For
         those elements, ordinary string formatting should be used before
         running execute.
+
+        References
+        ----------
+        http://initd.org/psycopg/docs/usage.html
         """
         with self._sql_executor(sql, sql_args) as pgcursor:
             result = pgcursor.fetchone()
 
         return result
 
-    def fetchall_with_types(self, sql, sql_args=None):
-        """Executes a fetchall SQL query with column information
+    def execute_fetchall(self, sql, sql_args=None):
+        """Executes a fetchall SQL query
 
         Parameters
         ----------
         sql : str
             The SQL query
-        sql_args : tuple or list, optional
+        sql_args : tuple, list or dict, optional
             The arguments for the SQL query
 
         Returns
         ------
         list of tuples
             The results of the fetchall query
-        dict
-            dictionary in the form of {column: type}
-
-        Raises
-        ------
-        QiitaDBExecutionError
-            If there is some error executing the SQL query
 
         Notes
         -----
-        from psycopg2 documentation, only variable values should be bound
+        From psycopg2 documentation, only variable values should be bound
         via sql_args, it shouldn't be used to set table or field names. For
         those elements, ordinary string formatting should be used before
         running execute.
+
+        References
+        ----------
+        http://initd.org/psycopg/docs/usage.html
         """
         with self._sql_executor(sql, sql_args) as pgcursor:
             result = pgcursor.fetchall()
-            types = {desc[0]: self.TYPE_CODES[desc[1]]
-                     for desc in pgcursor.description}
 
-        return result, types
+        return result
 
-    def fetchone_with_types(self, sql, sql_args=None):
-        """Executes a fetchone SQL query with column information
+    def _check_queue_exists(self, queue_name):
+        """Checks if queue `queue_name` exists in the handler
 
         Parameters
         ----------
-        sql : str
-            The SQL query
-        sql_args : tuple or list, optional
-            The arguments for the SQL query
-
-        Returns
-        -------
-        Tuple
-            The results of the fetchone query
-        dict
-            dictionary in the form of {column: type}
+        queue_name : str
+            The name of the queue
 
         Raises
         ------
-        QiitaDBExecutionError
-            if there is some error executing the SQL query
+        KeyError
+            If the queue `queue_name` does not exist in the handler
+        """
+        if queue_name not in self.queues:
+            raise KeyError("Queue '%s' does not exist" % queue_name)
```

Comment:
I think you're right, semantically your proposal makes more sense that the code that I have. I will change that.